packages/client/src/views/RegisterForm/review/ReviewSection.tsx

```diff
@@ -2174,15 +2079,22 @@ function fieldToReadOnlyFields(field: IFormField): IFormField {
   }
   return readyOnlyField
 }
-
 export const ReviewSection = connect(
-  (state: IStoreState) => ({
-    registerForm: getRegisterForm(state),
-    registrationSection: getBirthSection(state, 'registration'),
-    documentsSection: getBirthSection(state, 'documents'),
-    scope: getScope(state),
-    offlineCountryConfiguration: getOfflineData(state),
-    language: getLanguage(state)
-  }),
-  { goToPageGroup, writeDeclaration }
+  (state: IStoreState, { form }: { form: IForm }) => {
+    const registrationSection = form.sections.find(
+      ({ id }) => id === 'registration'
+    )
+    const documentsSection = form.sections.find(({ id }) => id === 'documents')
+    if (!registrationSection || !documentsSection) {
+      throw new Error('"registration" & "documents" are required sections')
+    }
```

Comment:
So could we assert these are found? Like `form.sections.find(..)!` and let the software break as we have the validation, this error should never occur

Reply:
hm not sure which way to go. either way works tbh

Reply:
yeah. maybe the runtime code could spawn a question that "why they can be undefined here?", but a ! might spawn a question that "has the dev been lazy here" :D

maybe the bulletproof solution would do some tricks to help TS to understand that the .find will always find.